### PR TITLE
Replace message factory to `MessageSettings`

### PR DESCRIPTION
**Pull request by vjik**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ✔️

### CODE REVIEW

File: src/DefaultMessageSettings.php [hunk 1]
```diff
@@ -0,0 +1,183 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Mailer;
+
+use DateTimeImmutable;
+
+final class DefaultMessageSettings
```
Comment (samdark) [inline src/DefaultMessageSettings.php:9]:
```suggestion
final class MessageSettings
```

File: src/MessageSettings.php [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+
+use DateTimeImmutable;
+
+/**
+ * `MessageSettings` provides default and extra message settings.
+ */
+final class MessageSettings
+{
+    /**
+     * @var array[]|null
```
Comment (samdark) [inline src/MessageSettings.php:15]: Need description.

File: src/MessageSettings.php [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+     * @psalm-var array<string,list<string>>|null
+     */
+    public readonly array|null $headers;
+
+    /**
+     * @var array[]|null
```
Comment (samdark) [inline src/MessageSettings.php:21]: Need description.

**Review by samdark — APPROVED** [review]

File: src/MessageSettings.php (hunk 2 above)
Comment (vjik) [reply src/MessageSettings.php]: done

File: src/MessageSettings.php (hunk 3 above)
Comment (vjik) [reply src/MessageSettings.php]: done

**Pull request merged**